### ISSUE TITLE
Possible memory leak issue

### ISSUE THREAD

**Issue by transmitterdan** [jongough/ocpn_draw_pi#284]
I want to run OpenCPN in debug mode under Windows with as many plugins as possible to see if there are any issues. I have run into a problem with this plugin. It seems that the plugin links to WXSVG.LIB that is built by the OpenCPN build process. That library implements globally:

IMPLEMENT_ABSTRACT_CLASS(wxSVGDocument, wxSvgXmlDocument)

When ocpn_draw_pi.dll loads the OS causes the global initialization to happen again. The wxWidgets libraries built in debug mode throw a wxAssert after the first time this object is created and within the wx code is this comment:
```
    // Using wxIMPLEMENT_DYNAMIC_CLASS() macro twice (which may happen if you
    // link any object module twice mistakenly, or link twice against wx shared
    // library) will break this function because it will enter an infinite loop
    // and eventually die with "out of memory" - as this is quite hard to
    // detect if you're unaware of this, try to do some checks here.
    wxASSERT_MSG( classTable->Get(m_className) == NULL,
```
I believe it is a bug to link to the same library twice within one program like this. But I admit that I understand less than all of you who built this great plugin. But I fear it may be causing problems based on the dire warning above. There are also discussions about this on StackExchange that suggest likewise.

A solution might be to access the needed SVG functions through the plugin interface rather than linking directly to the library in the plugin. But that may mean a lot of work.
 
Alternatively, I think copying the SVG library sources over to the plugin and using a different derived object name will also cure the problem.

**Comment by jongough** [comment]
I always have issues with debugging OCPN under windows. I get quite a few assert issues at startup, but then it works OK. I let others build the release versions because of that. I do not know of a current memory leak or contention in the 1.4 version.

I cannot investigate until Oct as am away cruising and don't have access to my build/test machines until I get back. Hope you can wait untill then. If not can you find a solution and make sure none of the other plugins is creating this problem?

Thanks
Jon

**Comment by transmitterdan** [comment]
Jon,

I can help you with all the asserts that prevent debugging in windows. I have a fork that I keep current with OpenCPN master branch. It has all the asserts fixed so they don't interfere with debugging. If you wish to use it for debugging your plugin let me know.

I will look into a solution. I think this is a low priority issue as probably there is not enough of a leak to cause anyone serious problems.

Regards,
Dan

**Comment by jongough** [comment]
Hi Dan,
   If you have a fix could you create a pull request with it in and I will have a look at implementing when I get home.
Thanks
Regards
Jon

**Comment by jongough** [comment]
Hi Dan,
   Give the latest version, 1.4.22, a go. The SVG libraries should now not cause an issue.

Regards
   Jon

**Comment by transmitterdan** [comment]
Jon,

I ran latest version from your git in debug mode. It found several wxASSERT errors but no memory leak complaints.

It seems that wxFormBuilder still has bugs that cause it to add invalid alignment flags to several items. It does not cause any real problem since wxWidgets ignores the unneeded flags.

Dan

**Comment by jongough** [comment]
Dan,
   I am currently not seeing any assert errors from OD (may be my setup), but I do get timer ones from OCPN. Could you send through the items you have found and I will update the version in the repository (or create a pull in git for me to see what you have done)? Didier has done some changes to help get rid of asserts and these have all be implemented into 1.4.22.

Thanks
Regards
   Jon
